src/components/Video.vue

```diff
@@ -1,13 +1,179 @@
 <template>
-    <div>
-        <h2>Media here</h2>
+  <div class="grey lighten-4">
+    <div v-show="!isReadyToPlay" class="py-8 text-center">
+      <div v-if="errorLoadingPreviews" class="mx-4">
+        <v-alert
+          dense
+          type="error"
+        >Some previews are not loaded</v-alert>
+      </div>
+      <div v-else>
+        <v-progress-circular
+          :size="50"
+          indeterminate
+          color="primary"
+        ></v-progress-circular>
+        <div class="mt-4">
+          Loading segments... ({{ Math.round(readyCount/totalCount*100) }}%)
+        </div>
+      </div>
     </div>
+    <div v-show="isReadyToPlay">
+      <video
+        v-for="url, i in urls"
+        :key="i"
+        :src="url"
+        ref="video"
+        v-show="currentIndex % totalCount === i"
+        @canplaythrough="updateReadyCount()"
+      ></video>
+    </div>
+  </div>
 </template>
 
 <script>
+import Debug from 'debug';
+
+const debug = Debug('app:video');
 
 export default {
   name: 'Video',
-  props: {},
+  data() {
+    return {
+      currentIndex: -1,
+      urls: [],
+      durations: [],
+      timeout: 0,
+      forceReadyUpdateValue: 0,
+      errorLoadingPreviews: false,
+    };
+  },
+  methods: {
+    video(index) {
```

Comment:
```suggestion
    getVideo(index) {
```
And all the uses of it